CHANGELOG.md

```diff
@@ -12,6 +12,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 - Combining filters with AND
 - User-defined filters
 - Concatination of Dataframes
+- Math for vector operations on multiple columns
```

Comment:
Will do 👌

Reply:
Thanks for your comments. I agree this can be reworked to accommodate more than math. However, I'd be very sorry to see the math-specific string flavors of `op` go, or to add more overhead to a simple operation such that it can no longer be performed succinctly.

## Counter-proposal:

What if instead we split `df.Math` into three different new methods:

1. The first would have signature `Arithmetic(resultcol string, op string, operandcols ...string) DataFrame`, which only takes the limited string ops ("+", "-", "*", "/", and "%"), and takes variadic operands like `df.Math` currently does. I'd also like `Arithmetic` to be allowed to coerce values, because its purpose is to make common operations as easy as possible, but more on that below.
2. The second would have signature something like `ElemMultiApply(resultcol string, op func(elements ...Element) Element, operandcols ...string) DataFrame`, where the user passes a variadic function on `Element`s and however many columns, and it gets applied without coercion.
3. The third would have signature something like `FloatMultiApply(resultcol string, op interface{}, operandcols ...string) DataFrame` and use the same techniques as in the current `df.Math` to support unary, binary, and trinary `op` functions on at least `float64` values (and I'd really like to be able to automatically convert the `int`s in mixed operand columns to `float64` as necessary to enable pleasant access to the `math` package on integers - but see below for coercion discussion). It would have to be able to support all three arities to be able to take any function from `math` directly.

## No coercision?
Is the request not to do automatic coercion a gota policy set in stone? I thought there was already automatic coercion in gota. `Capply` and `Rapply` in the readme say "casting the types as necessary", and the function I used to figure out what the output should be (int or float) was already there; I just moved it out of a function to make it accessible to `Math`.

Are you sure you wouldn't want it in, when it's only automatic coercion in one direction (`int` -> `float64`) and it only happens when the input columns are mismatched (at least one `float64` column among the operands)? I would personally _much_ prefer a _concise_ API with a few well-documented potential gotchas to a verbose API that makes me do extra work in the most common cases. Coercion is also how I managed to make it possible without much ceremony to pass any function from Go's `math` package in as `op` and have it correctly apply to columns of mixed type (they get detected and cast to float64 to be compatible, and the output is always float64).

There's also type coercion in Pandas and R, and people seem to be able to handle it. I think a nice pile of warnings in the documentation would suffice, at least for me, and what we get for it is agility and API clarity. (And the reason I'm using gota in the first place is because idiomatic Go doesn't let me express a complex high level thought succinctly enough to do it often.)

All of that said, I'm flexible here, and it's your project. :)

## FindElem

As for `FindElem`, I think I can just remove that without sacrificing much. I currently perform the same operation in my existing code with `df.Filter(...).Elem(0,1).Float()` which is succinct enough. (If we do add something that gives you only the first match later though, I'd suggest `First` or `FirstRow` rather than `Head`, because `df.head` in Pandas shows the first n, defaulting to 5.)

As an aside, for when there are many rows, I was thinking of adding `Index(columnname string) DataFrame` which would build an index of the values in that column to their row number, and if a user chose to build such an index up-front, anything that needed to search for a value (e.g., `Filter`) would make use of it to improve performance. That's still possible with the Filter-Elem-Float paradigm for looking up a value.